prdoc/pr_5774.prdoc

```diff
@@ -0,0 +1,13 @@
+title: Avoid unnecessary state reset of allowed_requests when no block requests are sent
+
+doc:
+  - audience: Node Dev
+    description: |
+      Previously, the state of `allowed_requests` will always be reset to the default value
+      even if there are no new block requests in the end. This could cause an edge cause that
+      `peer_block_request()` will early return next time when no ongoing block requests in fact.
```

Comment:
```suggestion
      Previously, the state of `allowed_requests` was always reset to the default
      even if there were no new block requests. This could cause an edge case
      because `peer_block_request()` will return early next time when there are no ongoing block requests.
```